test/conftest.py

```diff
@@ -37,10 +51,21 @@
 SDK_CALLBACK_URL = os.environ["SDK_CALLBACK_URL"]
 
 
+@pytest.fixture(scope="session")
+def data_fixture(request: FixtureRequest, ws_id: int) -> list[Any] | dict[str, Any]:
```

Comment:
One of the bits of pytest magic: this function allows you to request a fixture by name, but not just that -- having the `ws_id` parameter allows you to specify the workspace ID that the fixture should have been created under.

The SetAPI tests are run in two workspaces to allow testing of the `list_all_sets` function; the tests for that start off with no sets in the workspace and then rerun `list_all_sets` after adding some sets. Nearly all the tests use the `default` workspace, which has params `default_ws_id` and `default_ws_name`. The `list_all_sets` workspace is uses (surprisingly!) `list_all_sets_ws_id` and `list_all_sets_ws_name`.